soft_999_Jenish_Patel_fw_102

```diff
@@ -0,0 +1,45 @@
+// Blink an LED
+#include <stdint.h>
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+#define HALF_SECOND 500  //
+
+typedef struct CounterStorage {
+  uint16_t counter_a;
+  uint16_t counter_b;
+} CounterStorage;
+
+void prv_timer_call(const SoftTimerId timer_id, void *empty) {
+  // cast void* to our struct so we can use it
+  CounterStorage *storage = empty;
+  storage->counter_a++;
+  storage->counter_b++;
+
+  // Prints Every half-second
+  LOG_DEBUG("Counter A: %i\n", storage->counter_a);
+
+  // Prints Every Second
+  if (storage->counter_b % 2 == 0) {
+    LOG_DEBUG("Counter B: %i\n", storage->counter_b / 2);
```

Comment:
interesting solution, but yea this works. Could be confusing for someone to read though. I'm ok with this for here, but I think a better solution is to only increment every half second because it makes it more obvious what's happening.